dataops/components/resource_lock/views.py

```
@@ -0,0 +1,108 @@
+    level_stderr=ConfigClass.LOG_LEVEL_STDERR,
+).get_logger()
+
+router = APIRouter()
+
+
+@cbv(router)
+class RLock:
+    @router.post('/', response_model=ResourceLockResponseSchema, summary='Create a new lock')
+    @catch_internal('api_resource_lock')
+    async def lock(
+        self, body: ResourceLockCreateSchema, resource_locker: ResourceLockerCRUD = Depends(get_resource_lock)
+    ) -> ResourceLockResponseSchema:
+        unlocked = await resource_locker.perform_rw_lock(body.resource_key, body.operation)
+        response = ResourceLockResponseSchema(
+            code=EAPIResponseCode.success if unlocked else EAPIResponseCode.conflict, key=body.resource_key
+        )
+        return response.json_response()
+
+    @router.post('/bulk', response_model=ResourceLockBulkResponseSchema, summary='Create multiple locks')
+    @catch_internal('api_resource_lock')
+    async def bulk_lock(
+        self, body: ResourceLockBulkCreateSchema, resource_locker: ResourceLockerCRUD = Depends(get_resource_lock)
+    ) -> ResourceLockBulkResponseSchema:
+        lock_result = await resource_locker.perform_bulk_lock(body.resource_keys, body.operation)
+
+        response = ResourceLockBulkResponseSchema(
+            code=EAPIResponseCode.success if lock_result.is_successful() else EAPIResponseCode.conflict,
+            keys_status=lock_result.status,
+        )
+
+        return response.json_response()
+
+    @router.delete('/', response_model=ResourceLockResponseSchema, summary='Remove a lock')
+    @catch_internal('api_resource_lock')
+    async def unlock(
+        self, body: ResourceLockCreateSchema, resource_locker: ResourceLockerCRUD = Depends(get_resource_lock)
+    ) -> ResourceLockResponseSchema:
+        flag = await resource_locker.perform_rw_unlock(body.resource_key, body.operation)
+
+        response = ResourceLockResponseSchema(
+            code=EAPIResponseCode.success if flag else EAPIResponseCode.bad_request, key=body.resource_key
```

we should raise an HTTP error with the correct code. 
like this one:
https://github.com/PilotDataPlatform/dataops/blob/develop/dataops/components/exceptions.py#L74

---

Fixed. Now raising exceptions instead from https://github.com/PilotDataPlatform/dataops/blob/develop/dataops/components/exceptions.py.